### PR TITLE
Add DianaNicole

### PR DESCRIPTION
**Pull request by WarriorGallade**

### CODE REVIEW

File: data/mods/gen9ssb/conditions.ts [hunk 1]
```diff
@@ -616,6 +616,18 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			this.add(`c:|${getName('dhelmise')}|ok`);
 		},
 	},
+	diananicole: {
+		noCopy: true,
+		onStart() {
+			this.add(`c:|${getName('DianaNicole')}|Ready for Initiative? Cause I’m gonna Clickity Clackity, Roll to Attackity!`);
```
Comment (KrisXV) [inline data/mods/gen9ssb/conditions.ts:622]:
```suggestion
			this.add(`c:|${getName('DianaNicole')}|Ready for Initiative? Cause I'm gonna Clickity Clackity, Roll to Attackity!`);
```

File: data/mods/gen9ssb/moves.ts [hunk 2]
```diff
@@ -1534,6 +1534,56 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+				/* Problem here - we can't retroactively change the multihit parameter.
+				With this specific code, the move functions as intended, but will display the incorrect
+				number of hits if a target is immune to any of them. Even if you try to return false, null, etc
+				during this step, it will not interrupt the move. Nor will a this.add(-fail) do so either.
+				This seems to be the only way to get it to work and is a decent enough compromise for now. */
+			} else {
+				move.type = hitTypes[move.hit];
+				const moveAnims = ['Flamethrower', 'Ice Beam', 'Gunk Shot', 'Charge Beam', 'Sludge Bomb'];
+				const hitAnims = [];
+				for (let i = 0; i < moveAnims.length; i++) {
```
Comment (KrisXV) [inline data/mods/gen9ssb/moves.ts:1570]:
```suggestion
				for (const [anim, i] of moveAnims.entries()) {
```

File: data/mods/gen9ssb/moves.ts [hunk 3]
```diff
@@ -1534,6 +1534,56 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+				during this step, it will not interrupt the move. Nor will a this.add(-fail) do so either.
+				This seems to be the only way to get it to work and is a decent enough compromise for now. */
+			} else {
+				move.type = hitTypes[move.hit];
+				const moveAnims = ['Flamethrower', 'Ice Beam', 'Gunk Shot', 'Charge Beam', 'Sludge Bomb'];
+				const hitAnims = [];
+				for (const [anim, i] of moveAnims.entries()) {
+					const index2 = Math.min(i, hitTypes.length - 1);
+					if (moveTypes[i] === hitTypes[index2]) {
+						hitAnims.push(moveAnims[i]);
```
Comment (KrisXV) [inline data/mods/gen9ssb/moves.ts:1573]:
```suggestion
						hitAnims.push(anim);
```

**Review by WarriorGallade — COMMENTED** [review]
this change (and the earlier one to the loop syntax) seems to break the animations, so it'll likely need to be reverted

**Pull request merged**